File: assignments/springboot/jpa-board/src/main/java/com/example/jpaboard/board/controller/BoardController.java
package com.example.jpaboard.board.controller;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestParam;

@Controller
public class BoardController {

    @GetMapping("/")
    public String boardList() {
        return "board-list";
    }

    @GetMapping("/write")
    public String write() {
        return "board-write";
    }

    @GetMapping("/detail")
    public String boardDetail(
            @RequestParam Long id,
            Model model
    ) {
        model.addAttribute("id", id);

        return "board-detail";
    }

    @GetMapping("/update/{id}")
    public String update(
            @PathVariable long id,
            Model model
    ) {
        model.addAttribute("id", id);

        return "board-update";
    }
}

File: practice/springboot/board/kotlin-board/src/main/kotlin/com/example/kotlinboard/controller/BoardController.kt
package com.example.kotlinboard.controller

import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestParam

@Controller
class BoardController {
    @GetMapping("/")
    fun boardList(): String = "board-list"

    @GetMapping("/write")
    fun write(): String = "board-write"

    @GetMapping("/detail")
    fun detail(@RequestParam("id") id: Long, model: Model): String {
        model.addAttribute("id", id)
        return "board-detail"
    }

    @GetMapping("/update/{id}")
    fun update(@PathVariable("id") id: Long, model: Model): String {
        model.addAttribute("id", id)
        return "board-update"
    }
}